File: src/main/java/mods/eln/sim/nbt/NbtElectricalLoad.java
package mods.eln.sim.nbt;

import mods.eln.misc.INBTTReady;
import mods.eln.sim.ElectricalLoad;
import net.minecraft.nbt.NBTTagCompound;

public class NbtElectricalLoad extends ElectricalLoad implements INBTTReady {

    String name;

    public NbtElectricalLoad(String name) {
        super();
        this.name = name;
    }

    public void readFromNBT(NBTTagCompound nbttagcompound, String str) {
        setVoltage(nbttagcompound.getFloat(str + name + "Uc"));
        if (Double.isNaN(getVoltage())) setVoltage(0);
        if (getVoltage() == Float.NEGATIVE_INFINITY) setVoltage(0);
        if (getVoltage() == Float.POSITIVE_INFINITY) setVoltage(0);
    }

    public void writeToNBT(NBTTagCompound nbttagcompound, String str) {
        nbttagcompound.setFloat(str + name + "Uc", (float) getVoltage());
    }
}

File: src/main/java/mods/eln/sim/nbt/NbtElectricalLoad.kt
package mods.eln.sim.nbt

import mods.eln.misc.INBTTReady
import mods.eln.sim.mna.state.ElectricalLoad
import net.minecraft.nbt.NBTTagCompound

open class NbtElectricalLoad(name: String) : ElectricalLoad(), INBTTReady {

    init {
        this.name = name
    }

    override fun readFromNBT(nbttagcompound: NBTTagCompound, str: String) {
        u = nbttagcompound.getDouble(str + name + "Uc")
        if (java.lang.Double.isNaN(u)) u = 0.0
        if (u == Double.NEGATIVE_INFINITY) u = 0.0
        if (u == Double.POSITIVE_INFINITY) u = 0.0
    }

    override fun writeToNBT(nbttagcompound: NBTTagCompound, str: String) {
        nbttagcompound.setDouble(str + name + "Uc", u)
    }
}
